scripts/cargo-install-all.sh

```diff
@@ -137,12 +142,52 @@ for bin in "${BINS[@]}"; do
   binArgs+=(--bin "$bin")
 done
 
+dcouBinArgs=()
+for bin in "${DCOU_BINS[@]}"; do
+  dcouBinArgs+=(--bin "$bin")
+done
+
+source "$SOLANA_ROOT"/scripts/dcou-tainted-packages.sh
+
+excludeArgs=()
+for package in "${dcou_tainted_packages[@]}"; do
+  excludeArgs+=(--exclude "$package")
+done
+
 mkdir -p "$installDir/bin"
 
+# Some binaries (like the notable agave-ledger-tool) need to acitivate
+# the dev-context-only-utils feature flag to build.
+# Build those binaries separately to avoid the unwanted feature unification.
+# Note that `--workspace --exclude <dcou tainted packages>` is needed to really
+# inhibit the feature unification due to a cargo bug. Otherwise, feature
+# unification happens even if cargo build is run only with `--bin` targets
+# which don't depend on dcou as part of dependencies at all.
 (
   set -x
-  # shellcheck disable=SC2086 # Don't want to double quote $rust_version
-  "$cargo" $maybeRustVersion build $buildProfileArg "${binArgs[@]}"
+  # Make sure dcou is really disabled by peeking the (unstable) build plan
+  # output after turning rustc into the nightly mode with RUSTC_BOOTSTRAP=1.
+  # In this way, additional requirement of nightly rustc toolchian is avoided.
+  # Note that `cargo tree` can't be used, because it doesn't support `--bin`.
+  # shellcheck disable=SC2086 # Don't want to double quote $maybeRustVersion
+  if (RUSTC_BOOTSTRAP=1 \
```

Comment:
hmm, i think `RUSTC_BOOSTRAP` should be the last resort. it's undocumented (yet somewhat used in the wild) feature. ;) so, our nightly deps shouldn't be replaced unless there's strong motivation. from devops side, the disk usage of nightly or other is bothersome?

Also, the our use of nightly is detect upstream changes earlier. so, `RUSTC_BOOSTRAP` is slightly against the goal, because rustc stable branches are cut quite early, compared to the nightly.

Lastly, bisecting will be harder too.

Reply:
it's a very good point! let's keep it!